src/readbin/func_start.ml

```diff
@@ -0,0 +1,37 @@
+open Bap.Std
+open Core_kernel.Std
+open Or_error
+
+exception Bad_user_input
+
+let byteweight bin =
+  let tmp = Filename.temp_file "bw_" ".output" in
+  let cmd = Printf.sprintf "bap-byteweight find -x %s > %s" bin tmp in
+  let _ = Unix.system cmd in
+  Symbols.read_addrset tmp
+
+let usersource = Symbols.read_addrset
+
+let symbols bin =
```

Comment:
This function is a copy-paste of the previous with slight changes. Find out the difference and abstract them.